### PR TITLE
feat: created transaction page, added request to get transactions

### PR DESCRIPTION
**Pull request by ProggerOleg**
## Describe your changes

- I made layout for transactions page
- Made request to get transactions for certain user
Screenshot from 2023-12-26 10-38-34

Screenshot from 2023-12-25 18-46-16

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/jira/software/projects/CC/boards/1?selectedIssue=CC-191&text=trans)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/transactions/styles.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export const TransactionModalTitle = styled.div`
+  font-weight: ${typography.fontWeightSemiBold};
+  font-size: ${typography.h3};
+  color: ${PRIMARY.black};
+  width: 90%;
+  display: flex;
+  justify-content: flex-start;
+`;
+
+export const TransactionWrapper = styled.div``;
```
Comment (nadiia-tsytsylina) [inline src/components/transactions/styles.ts:25]: Do you need this empty div?

Comment (nadiia-tsytsylina) [reply src/components/transactions/styles.ts:25]: and in figma top-up and payment have opposite colors

image

**Review by Lilly-nn — APPROVED** [review]

**Review by nadiia-tsytsylina — APPROVED** [review]

**Comment by Lilly-nn** [comment]
image
just a suggestion here, maybe to make more space between the header and the component?

File: src/components/transactions/helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
+import { USER_ROLE } from 'src/constants';
+import { PRIMARY, SECONDARY } from '../../theme/colors';
+import { TransactionAmmount } from './styles';
+
+export const getTransactionColor = (type: string, role: string): string => {
+  switch (type) {
+    case 'Outcome':
```
Comment (lyudad) [inline src/components/transactions/helpers.tsx:9]: it should be constant

File: src/components/transactions/helpers.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+};
+export const getTransactionTitle = (
+  type: string,
+  role: string,
+  amount: number,
+  translate: (key: string) => string
+): React.ReactNode => {
+  switch (type) {
+    case 'Outcome':
```
Comment (lyudad) [inline src/components/transactions/helpers.tsx:24]: same

**Review by lyudad — APPROVED** [review]

**Pull request merged**